wp/plugin/wp.ml

```diff
@@ -18,149 +18,158 @@ module Typ = Extension.Type
-          the function cannot be found in the binaries, the analysis will \
-          fail."
+    ~doc:{|Determines which function to verify. WP verifies a single function,
+           though calling it on the `main' function along with the `inline'
+           option will analyze the whole program. If no function is specified
+           or the function cannot be found in the binary/binaries, the analysis
+           will fail.|}
 
 (* Arguments that determine which properties CBAT should analyze. *)
 
 let precond = Cmd.parameter Typ.string "precond"
-    ~doc:"A custom precondition in SMT-LIB format to be used during the \
-          analysis. If no precondition is specified, a trivial precondition \
-          of `true' will be used."
+    ~doc:{|Allows the introduction of assertions to the beginning of a query.
+           This allows pruning of possible models. Assertions are specified in
+           the smt-lib2 format. For comparative predicates, one may refer to
+           variables in the original and modified programs by appending the
+           suffix `_orig' and `_mod' to variable names in the smt-lib
+           expression. For example, `--precond="(assert (= RDI_mod
+           #x0000000000000003)) (assert (= RDI_orig #x0000000000000003))".' If
+           no precondition is specified, a trivial precondition of `true' will
+           be used.|}
 
 let postcond = Cmd.parameter Typ.string "postcond"
-    ~doc:"A custom postcondition in SMT-LIB format to be used during the \
-          analysis. If no postcondition is specified, a trivial \
-          postcondition of `true' will be used."
+    ~doc:{|Replaces the default postcondition with the user-specified one,
+           using the smt-lib2 format. Similar to `--precond', one may create
+           comparative postconditions on variables by appending `_orig' and
+           `_mod' to variable names. If no postcondition is specified, a
+           trivial postcondition of `true' will be used.|}
 
 let trip_asserts = Cmd.flag "trip-asserts"
-    ~doc:"If set, WP will look for inputs to the subroutine that would cause \
-          an __assert_fail or __VERIFIER_error to be reached."
+    ~doc:{|Looks for inputs to the subroutine that would cause an
+           `__assert_fail' or `__VERIFIER_error' to be reached.|}
 
 let check_null_derefs = Cmd.flag "check-null-derefs"
-    ~doc:"If set, the WP analysis will check for inputs that would result in \
-          dereferencing a NULL value. In the case of a comparative analysis, \
-          WP will check that the modified binary has no additional null \
-          dereferences in comparison with the original binary."
+    ~doc:{|Checks for inputs that would result in dereferencing a null value
+           during a memory read or write. In the case of a comparative
+           analysis, checks that the modified binary has no additional null
+           dereferences in comparison with the original binary.|}
```

Comment:
I would say "In the case of comparative analysis, checks that the modified binary has no additional paths through the function that contain null dereferences in comparison with the original binary." It's the extra *paths* with null dereferences that WP finds, not extra null dereferences.